pkg/state/state.go

```diff
@@ -87,25 +86,27 @@ type ReleaseSetSpec struct {
 // helmStateAlias is helm state alias
 type helmStateAlias HelmState
 
-func (hs HelmState) MarshalYAML() (interface{}, error) {
```

Comment:
@yxxhero Same for MarshalYAML. goccy/go-yaml does support InterfaceMarshaller and expect the MarshalYAML function to return a Go object that is then serialized into YAML rather than a raw byte array that is embedded into the resulting YAML document as-is.

https://github.com/goccy/go-yaml/blob/6cdefc42e112ac71cbe316e1eed264ea62f58e25/yaml.go#L28-L31

Reply:
HelmState's MarshalYAML was created for v1, and I feel like it can be adapted to the latest goccy/go-yaml interface, which I don't need to introduce for v0.

Reply:
Thanks for your reply! See https://github.com/helmfile/helmfile/pull/604#discussion_r1057009477

Reply:
@yxxhero Have you seen some tests failing due to diffs like the below?

```
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -17,6 +17,6 @@
        	            	 metadata:
        	            	-  name: foo-2
        	            	-  namespace: default
        	            	+name: foo-2
        	            	+namespace: default
        	            	 data:
        	            	-  bar: BAR
        	            	+bar: BAR
        	            	 ---
        	            	@@ -26,6 +26,6 @@
        	            	 metadata:
        	            	-  name: foo-1
        	            	-  namespace: default
        	            	+name: foo-1
        	            	+namespace: default
        	            	 data:
        	            	-  foo: FOO
        	            	+foo: FOO
```

I think this indicates that MarshalYAML changes from https://github.com/helmfile/helmfile/pull/604/commits/589576e0111687298089b2e0cb57701b3703eda7 is unnecessary.

Reply:
yeah. I saw it. and I tried to solve this issue. but it's because goccy/yaml logic issue. I already post an issue for this.